src/lib/Rule/Internal/RemoveInterfaceWithMethods.php

```diff
@@ -0,0 +1,151 @@
+    {
+        $this->removeInterfacesRector = $removeInterfacesRector;
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Remove Interface implementation with all its methods', [new ConfiguredCodeSample(
+            <<<'CODE_SAMPLE'
+                class SomeClassWithInterface implements InterfaceToRemove
+                {
+                    public function interfaceMethod(): array
+                    {
+                        return ['smth'];
+                    }
+                }
+            CODE_SAMPLE
+            ,
+            <<<'CODE_SAMPLE'
+                class SomeClassWithInterface
+                {
+                }
+            CODE_SAMPLE
+            ,
+            ['var_dump']
+        )]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return $this->removeInterfacesRector->getNodeTypes();
+    }
+
+    /**
+     * @param \PhpParser\Node\Stmt\Class_ $node
+     */
+    public function refactor(Node $node): ?int
+    {
+        if ($node->implements === []) {
+            return null;
+        }
+
+        foreach ($node->implements as $implement) {
+            if ($this->isNames($implement, $this->interfacesToRemove)) {
+                foreach ($this->interfacesToRemove as $interface) {
+                    $ref = new ReflectionClass($interface);
+                    $methods = array_map(static fn (ReflectionMethod $reflectionMethod) => $reflectionMethod->getName(), $ref->getMethods());
+
+                    // Remove method definition
+                    foreach ($node->stmts as $key => $stmt) {
+                        if ($stmt instanceof ClassMethod && $this->isNames($stmt, $methods)) {
+                            unset($node->stmts[$key]);
+                        }
+                    }
+
+                    // Remove method calls, if one of the arguments was removed method
+                    foreach ($node->getMethods() as $classMethod) {
+                        $nodeTraverser = new NodeTraverser();
+                        $nodeTraverser->addVisitor(new class($this->nodeNameResolver, $methods) extends NodeVisitorAbstract {
+                            private NodeNameResolver $nodeNameResolver;
+
+                            /** @var string[] */
+                            private array $methods;
+
+                            /**
+                             * @param string[] $methods
+                             */
+                            public function __construct(NodeNameResolver $nodeNameResolver, array $methods)
+                            {
+                                $this->nodeNameResolver = $nodeNameResolver;
+                                $this->methods = $methods;
+                            }
+
+                            public function leaveNode(Node $node)
```

Comment:
Missing return type or at least description if it's mixed.

Reply:
This comes from \PhpParser\NodeVisitor interface, I do not think there is need to duplicate it.